server/src/test/java/org/elasticsearch/discovery/DiscoveryDisruptionIT.java

```diff
@@ -138,8 +136,8 @@ public void testUnicastSinglePingResponseContainsMaster() throws Exception {
      * Test cluster join with issues in cluster state publishing *
      */
     public void testClusterJoinDespiteOfPublishingIssues() throws Exception {
-        List<String> nodes = startCluster(2, 1);
-
+        List<String> nodes = startCluster(2);
+        setMinimumMasterNodes(1);
```

Comment:
I think here we should explicitly start a master and a non-master node rather than starting two master-eligible nodes and then breaking the config like this.